src/charm.py

```diff
@@ -8,18 +8,206 @@
+    def _hydra_layer(self) -> Layer:
+        """Returns a pre-configured Pebble layer."""
+        layer_config = {
+            "summary": "hydra-operator layer",
+            "description": "pebble config layer for hydra-operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "entrypoint of the hydra-operator image",
+                    "command": f"hydra serve all --config {self._hydra_config_path} --dev",
+                    "startup": "enabled",
+                }
+            },
+            "checks": {
+                "version": {
+                    "override": "replace",
+                    "exec": {"command": "hydra version"},
+                },
+                "ready": {
+                    "override": "replace",
+                    "http": {"url": "http://localhost:4445/health/ready"},
+                },
+            },
+        }
+        return Layer(layer_config)
+
+    @property
+    def _config(self) -> str:
+        """Returns Hydra configuration."""
+        db_info = self._get_database_relation_info()
+
+        config = {
+            "dsn": f"postgres://{db_info['username']}:{db_info['password']}@{db_info['endpoints']}/postgres",
+            "log": {"level": "trace"},
+            "secrets": {
+                "cookie": ["my-cookie-secret"],
+                "system": ["my-system-secret"],
+            },
+            "serve": {
+                "admin": {
+                    "host": "localhost",
+                    "port": 4445,
+                },
+                "public": {
+                    "host": "localhost",
+                    "port": 4444,
+                },
+            },
+            "urls": {
+                "consent": "http://localhost:3000/consent",
+                "login": "http://localhost:3000/login",
+                "self": {
+                    "issuer": "http://localhost:4444/",
+                    "public": "http://localhost:4444/",
+                },
+            },
+        }
+
+        return yaml.dump(config)
+
+    def _get_database_relation_info(self) -> dict:
+        relation_id = self.database.relations[0].id
+        relation_data = self.database.fetch_relation_data()[relation_id]
+
+        return {
+            "username": relation_data["username"],
+            "password": relation_data["password"],
+            "endpoints": relation_data["endpoints"],
+        }
+
+    def _update_layer(self) -> None:
+        """Updates the Pebble configuration layer and config if changed."""
+        self.unit.status = MaintenanceStatus("Applying pebble layer")
+        self._container.add_layer(self._container_name, self._hydra_layer, combine=True)
+        logger.info("Pebble plan updated with new configuration")
+
+        # Push the config on first layer update to avoid PathError
+        self._container.push(self._hydra_config_path, self._config, make_dirs=True)
```

Comment:
You are pushing the configs every time, not only on the first layer update

Reply:
Thanks for catching that